doc/api/cluster.markdown

```diff
@@ -88,30 +116,338 @@ Node.js process and a cluster worker differs:
+This can occur when a worker exits gracefully, is killed, or is disconnected
+manually (such as when using `worker.disconnect()`).
+
+There can be a delay between the `'disconnect'` and `'exit'` events.  These
+events can be used, for instance, to detect when a working process is stuck in
+a cleanup process or if there are long-living connections.
+
+When triggered, the callback function will be passed the `Worker` instance
+representing the worker that has disconnected:
+
+    cluster.on('disconnect', (worker) => {
+      console.log(`The worker #${worker.id} has disconnected`);
+    });
+
+### Event: 'exit'
+
+* `worker` {Worker object}
+* `code` {Number} the exit code, if it exited normally.
+* `signal` {String} the name of the signal (eg. `'SIGHUP'`) that caused
+  the process to be killed.
+
+When any worker processes exits or is terminated for any reason, the cluster
+module will emit the `'exit'` event within the master process.
+
+This can be used, for instance, to restart a failing worker by calling
+`cluster.fork()` again.
+
+    cluster.on('exit', (worker, code, signal) => {
+      console.log('worker %d died (%s). restarting...',
+        worker.process.pid, signal || code);
+      cluster.fork();
+    });
+
+See [child_process event: 'exit'][].
+
+### Event: 'fork'
+
+* `worker` {Worker object}
+
+When a new worker is forked the cluster module will emit a `'fork'` event
+within the master process.
+
+This can be used, for instance, to log worker activity or to create timeouts
+for managing worker lifetime:
+
+    const timeouts = [];
+    function errorMsg() {
+      console.error('Something must be wrong with the connection ...');
+    }
+
+    cluster.on('fork', (worker) => {
+      timeouts[worker.id] = setTimeout(errorMsg, 2000);
+    });
+    cluster.on('listening', (worker, address) => {
+      clearTimeout(timeouts[worker.id]);
+    });
+    cluster.on('exit', (worker, code, signal) => {
+      clearTimeout(timeouts[worker.id]);
+      errorMsg();
+    });
+
+### Event: 'listening'
+
+* `worker` {Worker object}
+* `address` {Object}
+
+After calling the `listen()` event on a TCP server inside a worker, whenever
+the `'listening'` event is emitted by the server object, a corresponding
+`'listening'` event will also be emitted by the `cluster` module in the
+master process.
+
+The event handler is passed two arguments when invoked: `worker` and `address`.
+The `worker` argument is a reference to the `Worker` object, and the `address`
+argument is an object containing the properties `address`, `port`, and
+`addressType`. These are particularly useful if the worker is listening
+on more than one address.
+
+    cluster.on('listening', (worker, address) => {
+      console.log(
+        `Worker ${worker.id} is now connected to ` +
+        `${address.address}:${address.port}`);
+    });
+
+The `addressType` is one of:
+
+* `4` (TCPv4)
+* `6` (TCPv6)
+* `-1` (unix domain socket)
```

Comment:
'local domain' I think is what its called in the net docs (its not "unix domain" on windows).